### PR TITLE
Add support for ubuntu-advantage with autoinstall

### PR DESCRIPTION
**Pull request by ogayot**
Hello,

This PR adds support for Ubuntu Advantage token when running Subiquity with `--autoinstall`.
Also added a dummy token in the integration tests.

Thanks,
Olivier

refs FR-1652

### CODE REVIEW

File: autoinstall-schema.json [hunk 1]
```diff
@@ -260,6 +260,14 @@
                 }
             ]
         },
+        "ubuntu-advantage": {
+            "type": "object",
+            "properties": {
+                "token": {
+                    "type": "string"
```
Comment (dbungert) [inline autoinstall-schema.json:267]: I'm considering if we want to do crude token checks at the schema level.
For example, 30 chars, upper/lower/digits only.
If the token changes format we'd need to update but I'm OK with that.
What do you think?

Comment (ogayot) [reply autoinstall-schema.json:267]: Now that the format is defined, I was thinking at using:
```json
{
    "type": "string",
    "minLength": 24,
    "maxLength": 30,
    "pattern": "^C[1-9A-HJ-NP-Za-km-z]{23,29}$
}
```

but maybe it's overkill.

Comment (dbungert) [reply autoinstall-schema.json:267]: I don't think that check is overkill, except maybe the {23,29} is redundant with min/maxLength. Let's also give this a brief comment explaining the format at a basic level - the constant at the front and a pointer to https://pkg.go.dev/github.com/btcsuite/btcutil/base58#CheckEncode would be good

**Review by dbungert — APPROVED** [review]

**Review by dbungert — CHANGES_REQUESTED** [review]
sorry, meant to click request changes for the schema enforcement changes + requested comment

File: autoinstall-schema.json (hunk 1 above)
Comment (ogayot) [reply autoinstall-schema.json]: Done, thanks!
Added the pattern without the redundant check.
Also added the brief explanation in the "description" field of the schema.

**Review by dbungert — APPROVED** [review]

**Pull request merged**